File: tests/security/get_health_test.go
package security_test

import (
	"os"
	"testing"

	"github.com/yourorg/sectester"
)

var runner *sectester.SecRunner

func TestMain(m *testing.M) {
	// Initialize SecTester
	runner = sectester.NewSecRunner(sectester.Config{
		Hostname:  os.Getenv("BRIGHT_HOSTNAME"),
		ProjectID: os.Getenv("BRIGHT_PROJECT_ID"),
	})

	// Ensure runner is properly initialized
	if err := runner.Init(); err != nil {
		panic(err)
	}

	// Run tests
	code := m.Run()

	// Cleanup
	runner.Clear()

	os.Exit(code)
}

func TestGetHealth(t *testing.T) {
	t.Parallel()

	baseUrl := os.Getenv("BRIGHT_TARGET_URL")

	tests := []string{
		"csrf",
		"excessive_data_exposure",
		"http_method_fuzzing",
		"insecure_tls_configuration",
		"version_control_systems",
	}

	attackParamLocations := []sectester.AttackParamLocation{
		sectester.AttackParamLocation.HEADER,
	}

	threshold := sectester.Severity.CRITICAL
	timeout := 40 * 60 // 40 minutes in seconds

	if err := runner.CreateScan(sectester.ScanConfig{
		Tests:               tests,
		AttackParamLocations: attackParamLocations,
		Threshold:           threshold,
		Timeout:             timeout,
	}).Run(sectester.ScanRunConfig{
		Method: sectester.HttpMethod.GET,
		Url:    baseUrl + "/health",
	}); err != nil {
		t.Fatalf("Scan failed: %v", err)
	}
}

File: .brightsec/tests/get-health.test.ts
import { test, before, after } from 'node:test';
import { SecRunner } from '@sectester/runner';
import { AttackParamLocation, HttpMethod } from '@sectester/scan';

const timeout = 40 * 60 * 1000;
const baseUrl = process.env.BRIGHT_TARGET_URL!;

let runner!: SecRunner;

before(async () => {
  runner = new SecRunner({
    hostname: process.env.BRIGHT_HOSTNAME!,
    projectId: process.env.BRIGHT_PROJECT_ID!
  });

  await runner.init();
});

after(() => runner.clear());

test('GET /health', { signal: AbortSignal.timeout(timeout) }, async () => {
  await runner
    .createScan({
      tests: ['csrf', 'http_method_fuzzing', 'improper_asset_management', 'insecure_tls_configuration', 'xss'],
      attackParamLocations: [AttackParamLocation.HEADER],
      starMetadata: {
        code_source: "NeuraLegion/golang-realworld-example-app:main",
        databases: ["MySQL"],
        user_roles: { roles: [] }
      },
      poolSize: +process.env.SECTESTER_SCAN_POOL_SIZE || undefined
    })
    .setFailFast(false)
    .timeout(timeout)
    .run({
      method: HttpMethod.GET,
      url: `${baseUrl}/health`,
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Credentials': 'true',
        'Access-Control-Allow-Headers': 'Content-Type, Content-Length, Accept-Encoding, X-CSRF-Token, Authorization, accept, origin, Cache-Control, X-Requested-With',
        'Access-Control-Allow-Methods': 'POST, OPTIONS, GET, PUT, DELETE'
      }
    });
});